SU2_CFD/src/fluid/CSU2TCLib.cpp

```diff
@@ -1829,227 +1946,134 @@ void CSU2TCLib::DiffusionCoeffGY(){
-    R += Ru * rhos[iSpecies]/Density;
+    R += Ru / MolarMass[iSpecies] * rhos[iSpecies]/Density;
   }
 
   /*--- Mixture thermal conductivity via Gupta-Yos approximation ---*/
-  ThermalCond_tr = 0.0;
-  ThermalCond_ve = 0.0;
+  su2double ThermalCond_tr = 0.0;
+  su2double ThermalCond_ve = 0.0;
   for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
 
     /*--- Calculate molar concentration ---*/
-    Mi      = MolarMass[iSpecies];
-    mi      = Mi/Na;
-    gam_i   = rhos[iSpecies] / (Density*Mi);
-    denom_t = 0.0;
-    denom_r = 0.0;
+    const su2double Mi    = (MolarMass[iSpecies] + EPS);
+    const su2double mi    = Mi/Na;
+    const su2double gam_i = rhos[iSpecies] / (Density*Mi);
+    su2double denom_t = 0.0;
+    su2double denom_r = 0.0;
+    su2double denom_re = 0.0;
+
     for (jSpecies = 0; jSpecies < nSpecies; jSpecies++) {
-      Mj    = MolarMass[jSpecies];
-      mj    = Mj/Na;
-      gam_j = rhos[iSpecies] / (Density*Mj);
-      a_ij = 1.0 + (1.0 - mi/mj)*(0.45 - 2.54*mi/mj) / ((1.0 + mi/mj)*(1.0 + mi/mj));
+      const su2double Mj    = (MolarMass[jSpecies] + EPS);
+      const su2double mj    = Mj/Na;
+      const su2double gam_j = rhos[iSpecies] / (Density*Mj);
+      const su2double a_ij = 1.0 + (1.0 - mi/mj)*(0.45 - 2.54*mi/mj) / ((1.0 + mi/mj)*(1.0 + mi/mj));
 
-      /*--- Calculate the Omega^(0,0)_ij collision cross section ---*/
-      Omega_ij = 1E-20 * Omega00(iSpecies,jSpecies,3)
-          * pow(T, Omega00(iSpecies,jSpecies,0)*log(T)*log(T)
-          + Omega00(iSpecies,jSpecies,1)*log(T)
-          + Omega00(iSpecies,jSpecies,2));
+      const su2double T_col = ((iSpecies == 0 && ionization) || (jSpecies == 0 && ionization)) ? Tve : T; 
 
-      /*--- Calculate "delta1_ij" ---*/
-      d1_ij = 8.0/3.0 * sqrt((2.0*Mi*Mj) / (pi*Ru*T*(Mi+Mj))) * Omega_ij;
+      su2double d1_ij = ComputeCollisionDelta(iSpecies, jSpecies, Mi, Mj, T_col, true);
+      su2double d2_ij = ComputeCollisionDelta(iSpecies, jSpecies, Mi, Mj, T_col, false);
+      if (d1_ij >1E16) d1_ij = 1E16;
+      if (d2_ij >1E16) d2_ij = 1E16;      
+
+      if (jSpecies == 0 && ionization) { denom_t += 3.54*gam_j*d2_ij; }
+      else { denom_t += a_ij*gam_j*d2_ij; }
 
-      /*--- Calculate the Omega^(1,1)_ij collision cross section ---*/
-      Omega_ij = 1E-20 * Omega11(iSpecies,jSpecies,3)
-          * pow(T, Omega11(iSpecies,jSpecies,0)*log(T)*log(T)
-          + Omega11(iSpecies,jSpecies,1)*log(T)
-          + Omega11(iSpecies,jSpecies,2));
-
-      /*--- Calculate "delta2_ij" ---*/
-      d2_ij = 16.0/5.0 * sqrt((2.0*Mi*Mj) / (pi*Ru*T*(Mi+Mj))) * Omega_ij;
-      denom_t += a_ij*gam_j*d2_ij;
       denom_r += gam_j*d1_ij;
+      denom_re += gam_j*d2_ij;
     }
 
-    /*--- Translational contribution to thermal conductivity ---*/
-    ThermalCond_tr += (denom_t > EPS) ? ((15.0/4.0)*kb*gam_i/denom_t) : su2double(0.0);
+    /*--- Prevent divide by 0 ---*/
+    if (denom_t == 0.0) denom_t = EPS;
```

Comment:
@pcarruscag @jtneedels This helps prevent the solution going NaN and seems to have little impact on solution quality/convergence behavior.   Is there a better/safer way to do this?

Reply:
LGTM - we can't say if denom < EPS because these values are so small (that was causing issues with the original implementation) so I think this is good.

Reply:
Is denom always positive? If so consider using fmax(denom, EPS) comparing floats with == can cause surprises.

Reply:
Unfortunately, adding in the fmax causes issues down the line in the simulation.  I added a check for denom<= 0 to avoid any extra weirdness.